compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -2259,12 +2259,12 @@ fn add_native_libs_from_crate(
     link_static: bool,
     link_dynamic: bool,
 ) {
-    if !sess.opts.unstable_opts.link_native_libraries {
-        // If `-Zlink-native-libraries=false` is set, then the assumption is that an
-        // external build system already has the native dependencies defined, and it
-        // will provide them to the linker itself.
-        return;
-    }
+    // If `-Zlink-native-libraries=false` is set, then the assumption is that an
+    // external build system already has the native dependencies defined, and it
+    // will provide them to the linker itself.
+    // The option doesn't currently apply to native libraries bundled into rlibs, regardless
+    // of the value of `-Zpacked-bundled-libs`, this may be changed in the future.
+    let link_native = sess.opts.unstable_opts.link_native_libraries;
```

Comment:
This commit goes against the purpose of `-Zlink-native-libraries`, doesn't it?

Reply:
Maybe, but I'm not sure.
If a crate brings some private C code with it, compiles it, and then bundles it, does it concern the outer build system?
In case of native libs bundled in the old way (as object files) we are not "unbundling" them if `-Zlink-native-libraries=false` is used, this commit at least makes the new behavior consistent with that.
`-Zpacked-bundled-libs` is unstable, so all of this is up to discussion.

Reply:
I think `-Zlink-native-libraries` only makes sense if the build system invoking rustc is also the one invoking the linker. As such this build system would know about the native lib bundled with the rlib. If these build systems are not the same, it is almost guaranteed to break as crates can add dependencies that the linking build system doesn't know about.

Reply:
I think I'll remove this commit after all.
Libraries bundled in the old way are added as object files so some third party build system linking rlibs as classic staticlibs will do more or less the right thing implicitly.
But libraries bundled in the old way will be ignored and not added to the link if rlibs are treated as staticlibs, so the build system will need to do something to get them linked anyway.

Reply:
I've removed this change.